hivemq-edge/src/frontend/src/extensions/datahub/components/pages/DataHubListings.tsx

```diff
@@ -1,19 +1,69 @@
-import { FC } from 'react'
+import { FC, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { Tab, TabList, TabPanel, TabPanels, Tabs, Text } from '@chakra-ui/react'
+import { Tab, TabList, TabPanel, TabPanels, Tabs, Text, useDisclosure, useToast } from '@chakra-ui/react'
+import { UseMutateAsyncFunction } from '@tanstack/react-query'
+
+import ConfirmationDialog from '@/components/Modal/ConfirmationDialog.tsx'
+
 import PolicyTable from '@datahub/components/pages/PolicyTable.tsx'
 import SchemaTable from '@datahub/components/pages/SchemaTable.tsx'
 import ScriptTable from '@datahub/components/pages/ScriptTable.tsx'
+import { dataHubToastOption } from '@datahub/utils/toast.utils.ts'
+
+interface DeleteMutationRequest {
+  mutation: UseMutateAsyncFunction<void, unknown, string, unknown>
+  type: string
+  id: string
+}
+
+export interface DataHubTableProps {
+  onDeleteItem?: (mutation: UseMutateAsyncFunction<void, unknown, string, unknown>, type: string, id: string) => void
+}
 
 const DataHubListings: FC = () => {
   const { t } = useTranslation('datahub')
+  const { isOpen: isConfirmDeleteOpen, onOpen: onConfirmDeleteOpen, onClose: onConfirmDeleteClose } = useDisclosure()
+  const [deleteItem, setDeleteItem] = useState<DeleteMutationRequest | undefined>(undefined)
+  const toast = useToast()
+
+  const handleConfirmOnClose = () => {
+    onConfirmDeleteClose()
+    setDeleteItem(undefined)
+  }
+
+  const handleConfirmOnSubmit = () => {
+    console.log(deleteItem)
+    if (!deleteItem?.mutation) return
```

Comment:
Does this mean that an item has been mutated?

Reply:
Not yet, just that the mutation function has not been defined